### PR TITLE
[tag explorer] Clicking on the same series should untoggle it

### PR DESCRIPTION
**Pull request by pavelpashkovsky**
## Brief
- Fixes this issue: https://github.com/pyroscope-io/pyroscope/issues/1377
    - clicking on the same series should untoggle it
    - when tag selected other tags shadowed
## Changes
- [DEMO](https://pr-1385.pyroscope.io/explore?query=rideshare-app-java.lock_count%7B%7D&groupBy=REGION&groupByValue=All)

https://user-images.githubusercontent.com/7372044/184339138-13cdb905-25a9-4876-998a-9f6cf5ce7df0.mp4

## Concerns
- `-`

### CODE REVIEW

**Comment by eh-am** [comment]
Is it possible to indicate visually that item has been toggled? Fading the other unselected ones maybe?

https://user-images.githubusercontent.com/6951209/184163733-d581fdcc-589c-4951-8ac6-87cadbcec79c.mp4

File: webapp/javascript/pages/tagExplorer/components/Legend.tsx [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
     <div data-testid="legend" className={styles.legend}>
       {groups.map(({ tagName, color }) => (
         <div
+          aria-hidden
           data-testid="legend-item"
           className={styles.tagName}
           key={tagName}
-          onClick={() => handleGroupByTagValueChange(tagName)}
+          onClick={() =>
+            handleGroupByTagValueChange(
```
Comment (Rperry2174) [inline webapp/javascript/pages/tagExplorer/components/Legend.tsx:27]: I'm not sure whats going on here can you explain / make this code a littler clearer? 

for example could this be simplified to a variable `isSelected`

File: webapp/javascript/pages/tagExplorer/components/Legend.tsx [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
+              tagName === activeGroup
```
Comment (Rperry2174) [inline webapp/javascript/pages/tagExplorer/components/Legend.tsx:28]: I'm not sure whats going on here can you explain / make this code a littler clearer? 

for example could this be simplified to a variable `isSelected`

Comment (eh-am) [reply webapp/javascript/pages/tagExplorer/components/Legend.tsx:28]: Also what is `appWithoutTagsWhereDropdownOptionName`?

File: webapp/javascript/pages/tagExplorer/components/Legend.tsx [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
+function Legend({
+  groups,
+  handleGroupByTagValueChange,
+  activeGroup,
+}: LegendProps) {
   return (
     <div data-testid="legend" className={styles.legend}>
       {groups.map(({ tagName, color }) => (
         <div
+          aria-hidden
```
[the remark this answers is not in the document]
Comment (pavelpashkovsky) [reply webapp/javascript/pages/tagExplorer/components/Legend.tsx:22]: Project linter has warning rule `[click-events-have-key-events]`(https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/bb84abc793435a25398160242c5f2870b83b72ca/docs/rules/click-events-have-key-events.md). Maybe we can disable (allow having interactable elements without key handlers) this rule on global level?
image 

or I can remove this `aria-hidden`

File: webapp/javascript/pages/tagExplorer/components/Legend.tsx (hunk 2 above)
Comment (pavelpashkovsky) [reply webapp/javascript/pages/tagExplorer/components/Legend.tsx]: @Rperry2174, refactored this part. @eh-am , I used existing constant, but I'm not sure that it's a good name, I renamed in to `ALL_TAGS` (I can revert it if you don't like)

**Review by Rperry2174 — APPROVED** [review]
LGTM!

**Pull request merged**